google-cloud-storage/src/main/java/com/google/cloud/storage/GrpcStorageOptions.java

```diff
@@ -97,35 +99,51 @@ StorageSettings getStorageSettings() throws IOException {
     StorageSettings.Builder builder =
         StorageSettings.newBuilder()
             .setEndpoint(endpoint)
-            .setCredentialsProvider(credentialsProvider);
+            .setCredentialsProvider(credentialsProvider)
+            .setClock(getClock());
 
     if (scheme.equals("http")) {
       builder.setTransportChannelProvider(
-          com.google.api.gax.grpc.InstantiatingGrpcChannelProvider.newBuilder()
+          InstantiatingGrpcChannelProvider.newBuilder()
               .setEndpoint(endpoint)
               .setChannelConfigurator(ManagedChannelBuilder::usePlaintext)
               .build());
     }
-    RetrySettings retrySettings = getRetrySettings();
-    RetrySettings attemptOnce = retrySettings.toBuilder().setMaxAttempts(1).build();
+    RetrySettings readRetrySettings =
+        getRetrySettings()
+            .toBuilder()
+            // when performing a read via ReadObject, the ServerStream will have a default relative
+            // deadline set of `requestStartTime() + totalTimeout`, meaning if the specified
+            // RetrySettings have a totalTimeout of 10 seconds -- which should be plenty for
+            // metadata RPCs -- the entire ReadObject stream would need to complete within 10
+            // seconds.
+            // To allow read streams to have longer lifespans, crank up their timeouts, instead rely
+            // on idleTimeout below.
+            .setLogicalTimeout(Duration.ofDays(28))
```

Comment:
Why 28 days and not 24 hours? Don't mean to bikeshed, but i don't follow what makes a sensible default here.

Reply:
🤷🏻 24 hours felt like it could bump up against some slow workload while 28 days is longer than most anyone would be willing to wait for even a slow system.

Reply:
fair enough; it's the "turbo" long-time for me is all. Please ignore.